src/utils.jl

```diff
@@ -1128,3 +1128,56 @@ _argument_description(s::Symbol) =
 
 _document_argument(s::Symbol) =
     _fmt_paragraph(_argument_description(s), leadingspaces = 6 + length(string(s)))
+
+#
+# The following functions implement the guess of the optimal legend position,
+# from the data series.
+#
+# Computes the distances of the plot limits to a sample of points at the extremes of 
+# the ranges, and places the legend at the corner where the maximum distance to the limits is found.
+#
+function _dmin_point(dmin, x, y, lim, scale)
+    p_scaled = (x / scale[1], y / scale[2])
+    d = sum(abs2, lim .- p_scaled)
+    dmin = ifelse(d < dmin, d, dmin) # preferred relative to min because x < NaN === false 
+    return dmin
+end
+function _dmin_series(lim, scale, x, y, nsamples)
+    length(x) > 0 || return +Inf
+    dmin = +Inf
+    step = 1
+    lim = lim ./ scale
+    n = max(1, div(min(nsamples, length(x)), 2))
+    # Run from the extremes of the dataset inwards
+    for isample in firstindex(x):step:n
+        dmin = _dmin_point(dmin, x[isample], y[isample], lim, scale)
+    end
+    for isample in lastindex(x):(-step):(lastindex(x) - n)
+        dmin = _dmin_point(dmin, x[isample], y[isample], lim, scale)
+    end
+    return dmin
+end
+#
+# Function barrier because lims are type-unstable
+#
+function _guess_best_legend_position(xl, yl, plt, nsamples)
+    scale = (maximum(xl) - minimum(xl), maximum(yl) - minimum(yl))
```

Comment:
```suggestion
    scale = (maximum(xl) - minimum(xl), maximum(yl) - minimum(yl))
```

Isn't it faster to use `extrema` here (one single pass) ?

Reply:
Doesn't make any difference, and makes the code uglier, IMO (have do add two lines to first compute the extrema and then use it)